src/lib.rs

```diff
@@ -984,7 +1027,7 @@ impl<'a, 'input: 'a> Node<'a, 'input> {
     /// Returns the first child of this node.
     #[inline]
     pub fn first_child(&self) -> Option<Self> {
-        self.d.children.map(|(id, _)| self.gen_node(id))
+        self.d.last_child.map(|_| self.gen_node(self.id + 1))
```

Comment:
Can't we use just `self.gen_node(self.id + 1)`?

Reply:
We can't. It addresses the question you posed: 
> How do we know that id+1 is actually a first child and not a sibling?

We know it's a first child and not a sibling because we check to see if the element has a last child. If it has a last child, it must also have a first child @ `self.id + 1`, if there is no last child, then `self.id + 1` points to the same node as `next_subtree` (which in most cases would be the next sibling)